src/serviceClients/jobManagerWrapper.ts

```diff
@@ -67,7 +88,35 @@ export class JobManagerWrapper extends JobManagerClient {
         };
       }),
     };
+    const res = await this.post<ICreateJobResponse>(createLayerTasksUrl, createJobRequest);
+    return res;
+  }
 
+  private async createUpdateJob(
```

Comment:
why do you duplicate the create job method? 
you can use same method with different params, no?

Reply:
no, the data type is diferent, update gets the version too. and NewRasterLayer is different from UpdateRasterLayer and i dont want to check types

Reply:
createInitJob(data: NewRasterLayer)
createInitUpdateJob(
    id: string,
    version: string,
    internalId: string,
    data: UpdateRasterLayer,
    updateJobAction: UpdateJobAction
    
    the params are completely different, UpdateRasterLayer doesnt hold the metadata that NewRasterLayer holds and the filling of CreateJobBody is completely different

Reply:
moreover, the jobs that are created are different - you can check the createJobRequest in each function

Reply:
this is the new job:
{
      resourceId: data.metadata.productId,
      version: '1.0',
      type: jobType,
      status: OperationStatus.PENDING,
      parameters: { ...data } as unknown as Record<string, unknown>,
      productName: data.metadata.productName,
      productType: data.metadata.productType,
      domain: this.jobDomain,
      tasks: taskParams?.map((params) => {
        return {
          type: taskType,
          parameters: params,
        };
      }),
    }
    
    this is the update:
     {
      resourceId: id,
      version: version,
      internalId: internalId,
      type: jobType,
      status: OperationStatus.PENDING,
      parameters: { ...data } as unknown as Record<string, unknown>,
      domain: this.jobDomain,
      tasks: taskParams?.map((params) => {
        return {
          type: taskType,
          parameters: params,
        };
      }),
    }
    
    _new has:_      
     productName: data.metadata.productName,
     productType: data.metadata.productType,
    _update has:_ 
    internalId: internalId (no name or type - technically can be added - needs to be sent from ingestion manger but are not necessary)
    
     i think that that is an overkill to try to make one function with so many open and possible parameters

Reply:
agreed with shlomi to keep it this way...